js/TableDetailComponent.js

```diff
@@ -99,6 +134,66 @@ class TableDetailComponent {
 
         // Close panel
         this.$detailPanel.removeClass('table-detail--open');
+
+        // Hide panel contents from screen readers
+        this.$detailPanel.attr('aria-hidden', 'true');
+
+        // Make sure focusable elemnts cannot gain focus whilst panel is closed
+        this.$detailPanel
+            .find('a[href], area[href], input, select, textarea, button, iframe, object, embed, [tabindex], *[contenteditable]')
+            .not('[tabindex=-1], [disabled], :hidden, [aria-hidden]')
+            .attr('tabindex', '-1');
+
+        // Remove previously bound listener
+        this.$html.off('keydown', this.boundKeydownListener);
+    }
+
+    /**
+     * Listen for keyboard navigation, trap tabbing within the panels focusable elements
+     * @param {jQuery} $focusableElements - Collection of focusable elements in the details panel
+     */
+    keydownListener ($focusableElements, event) {
+        let keyCode = event.keyCode || event.which;
```

Comment:
The general consensus is that `let` should be used in contexts where the variable will be reassigned. In situations where a variable is declared and assigned once, it should probably be a `const`.

There is an interesting argument / quite sweary rant (https://jamie.build/const) that you should only ever use `let` because javascript "constants" can be mutated, I can understand the principle, but I'm not sure I agree with it. I think the distinction between `let` and `const` is useful because when skimming code it's easy to reason about which variables may end up reassigned.